openpype/hosts/blender/plugins/publish/extract_blend.py

```diff
@@ -50,17 +50,19 @@ def process(self, instance):
 
         # Set object mode
         with plugin.context_override(
-            active=next(  # Get first object that is not library override
+            # Get first visible object that is not library override
+            active=next(
                 (
                     o
                     for o in bpy.context.scene.objects
-                    if o.override_library is None
+                    if o.override_library is None and o.visible_get()
                 ),
                 None,
             ),
             selected=bpy.context.scene.objects,
         ):
-            bpy.ops.object.mode_set()
+            if bpy.ops.object.mode_set.poll():  # last safty check
```

Comment:
I hesitated a long time to do this, but what if poll fails and an object is still in pose mode?

Reply:
Yes you're right, put this condition right before the `with` statement to avoid useless operation if not needed.